### PR TITLE
Bug: Clicking push notification doesn't navigate to the correct screen (KIDS-1735)

### PR DESCRIPTION
**Pull request by TammiLion**
## Release Notes

- **New Features**
	- Enhanced navigation capabilities within the family app, allowing users to switch tabs based on profile selection.
	- Integrated Firebase Cloud Messaging for improved notification handling and navigation based on received messages.

- **Bug Fixes**
	- Simplified routing logic by removing unnecessary redirects, improving user experience.

- **Improvements**
	- Refined state management for the navigation bar to ensure accurate UI updates based on user interactions and notifications.

### CODE REVIEW

**Review by Daniela510 — APPROVED** [review]

File: lib/features/give/pages/home_page.dart [hunk 1]
```diff
@@ -56,6 +57,13 @@ class _HomePageState extends State<HomePage> {
     super.initState();
     WidgetsBinding.instance.addPostFrameCallback((_) {
       context.read<InfraCubit>().checkForUpdate();
+
+      FirebaseMessaging.instance.getInitialMessage().then((message) {
+        final auth = context.watch<AuthCubit>().state;
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/give/pages/home_page.dart:62]: fair?

File: lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart [hunk 2]
```diff
@@ -210,6 +219,8 @@
     switch (custom) {
       case BoxOriginDialog():
         await _showBoxOriginModal(context);
+      case final SwitchTab event:
+        _setIndex(event.tabIndex);
```
[the remark this answers is not in the document]
Comment (TammiLion) [reply lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart:223]: Good suggestion.

File: lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import 'dart:async';
 
+import 'package:firebase_messaging/firebase_messaging.dart';
```
[the remark this answers is not in the document]
Comment (TammiLion) [reply lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart:3]: Good bot.

[the remark this answers is not in the document]
Comment (TammiLion) [reply lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart:3]: The current code is a future not a streamsubscription. This comment doesn't apply.

**Pull request merged**